haystack/document_stores/elasticsearch.py

```diff
@@ -28,6 +28,9 @@
 from haystack.document_stores.filter_utils import LogicalFilterClause
 from haystack.errors import DocumentStoreError, HaystackError
 
+if TYPE_CHECKING:
+    from haystack.nodes.retriever import DenseRetriever
```

Comment:
I just used the same imports as for `BaseRetriever` before. I supposed this was because of circular imports yes. The regular import works however, so I converted them. 
I could track the introduction of this line to [this PR](https://github.com/deepset-ai/haystack/pull/1624). It has a very suspicious commit with message "Fix circular import on BaseRetriever", so probably this has been fixed over the time :-)